### PR TITLE
[Dashboard] Replace Chakra-ui's useClipboard

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to add the `useClipboard` hook to various components and update existing imports.

### Detailed summary
- Added `useClipboard` hook to multiple components
- Updated imports for `useClipboard` hook
- Created `useClipboard` hook in `useClipboard.ts` file

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by kien-ngo** [comment]
* **#4660** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4660?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @kien-ngo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/@3rdweb-sdk/react/hooks/useClipboard.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { useCallback, useEffect, useState } from "react";
+
+/**
+ * todo: this hook is copied from the SDK, so might as well expose it from the SDK
+ * and use it here?
+ */
+export function useClipboard(text: string, delay?: number) {
```
Comment (jnsdls) [inline apps/dashboard/src/@3rdweb-sdk/react/hooks/useClipboard.ts:7]: can set the default here `delay = 1500`

File: apps/dashboard/src/@3rdweb-sdk/react/hooks/useClipboard.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  // legitimate usecase
+  // eslint-disable-next-line no-restricted-syntax
+  useEffect(() => {
+    let timeoutId: number | null = null;
+
+    if (hasCopied) {
+      timeoutId = window.setTimeout(() => {
+        setHasCopied(false);
+      }, delay || 1500);
```
Comment (jnsdls) [inline apps/dashboard/src/@3rdweb-sdk/react/hooks/useClipboard.ts:23]: and then remove the `|| 1500` here

File: apps/dashboard/src/@3rdweb-sdk/react/hooks/useClipboard.ts
Comment (jnsdls) [inline apps/dashboard/src/@3rdweb-sdk/react/hooks/useClipboard.ts:1]: don't add new files in `@3rdweb-sdk/...` we need to slowly get rid of it

feel free to add it into `src/hooks` instead

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/hooks/useClipboard.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { useCallback, useEffect, useState } from "react";
+
+/**
+ * todo: this hook is copied from the SDK, so might as well expose it from the SDK
+ * and use it here?
+ */
```
Comment (jnsdls) [inline apps/dashboard/src/hooks/useClipboard.ts:6]: we can consider that but it feels very "random utility"-esque - this is fine for now

**Review by jnsdls — APPROVED** [review]

**Pull request merged**